flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/ResourceProfileInfo.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.rest.messages;
+
+import org.apache.flink.api.common.resources.Resource;
+import org.apache.flink.runtime.clusterframework.types.ResourceProfile;
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonCreator;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Contains information of a {@link ResourceProfile}.
+ */
+public class ResourceProfileInfo {
+
+	public static final String FIELD_NAME_CPU = "cpuCores";
+
+	public static final String FIELD_NAME_TASK_HEAP = "taskHeapMemory";
+
+	public static final String FIELD_NAME_TASK_OFFHEAP = "taskOffHeapMemory";
+
+	public static final String FIELD_NAME_MANAGED = "managedMemory";
+
+	public static final String FIELD_NAME_NETWORK = "networkMemory";
+
+	public static final String FIELD_NAME_EXTENDED = "extendedResources";
+
+	@JsonProperty(FIELD_NAME_CPU)
+	private final double cpu;
+
+	@JsonProperty(FIELD_NAME_TASK_HEAP)
+	private final int taskHeapMB;
+
+	@JsonProperty(FIELD_NAME_TASK_OFFHEAP)
+	private final int taskOffHeapMB;
+
+	@JsonProperty(FIELD_NAME_MANAGED)
+	private final int managedMB;
+
+	@JsonProperty(FIELD_NAME_NETWORK)
+	private final int networkMB;
+
+	@JsonProperty(FIELD_NAME_EXTENDED)
+	private final Map<String, Double> extendedResources;
+
+	@JsonCreator
+	public ResourceProfileInfo(
```

Comment:
We should add a `RestResponseMarshallingTestBase` for the `ResourceProfileInfo`. See `TaskManagerInfoTest` for example.